tests/test_ess_csc.py

```diff
@@ -91,17 +91,15 @@ async def test_bin_script(self):
     async def validate_telemetry(self):
         for sensor_name in self.csc.device_configurations:
             device_configuration = self.csc.device_configurations[sensor_name]
-            telemetry_topic = getattr(
-                self.remote, f"tel_temperature{device_configuration.num_channels}Ch"
-            )
-            data = await telemetry_topic.next(flush=False)
+            data = await self.remote.tel_temperature.next(flush=False)
+            self.assertEquals(data.numChannels, device_configuration.num_channels)
+            temperature = data.temperature
             for i in range(0, device_configuration.num_channels):
-                # Channels indices are 1-based
-                temp_telemetry = getattr(data, f"temperatureC{i + 1:02d}")
-                self.assertLessEqual(common.MockTemperatureConfig.min, temp_telemetry)
-                self.assertLessEqual(temp_telemetry, common.MockTemperatureConfig.max)
+                self.assertLessEqual(common.MockTemperatureConfig.min, temperature[i])
+                self.assertLessEqual(temperature[i], common.MockTemperatureConfig.max)
```

Comment:
You might consider avoiding the loop with something like the following. I admit it has to process every element instead of short-circuiting. numpy.testing probably has an assert that short-circuits, if you care.
```
temperature = data.temperature[: device_configuration.num_channels]
assert common.MockTemperatureConfig.min <= min(temperature)
assert common.MockTemperatureConfig.max >= max(temperature)
```

Reply:
I gently suggest moving towards bare `assert` in unit tests. I think it makes the test code easier to read The output from `pytest` is just fine (unlike the output if you run the test using `python`. We never do that and I have been removing the `if __main__ ...` boilerplate from the end of my unit tests).